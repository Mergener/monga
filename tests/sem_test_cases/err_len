type T = {
    i : int;
};

function main() {
    var t : T;

    @len(5); # not ok, 5 is not indexable
    @len(t); # not ok, t is not indexable
    @len(new T[24]); # ok
    @len(new T); # not ok, t is not indexable
}