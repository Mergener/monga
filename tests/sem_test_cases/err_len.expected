[Error] Module tests/sem_test_cases/err_len, line 8, column 12: Invalid unary operation for type 'int'.
[Error] Module tests/sem_test_cases/err_len, line 9, column 12: Invalid unary operation for type 'T'.
[Error] Module tests/sem_test_cases/err_len, line 11, column 16: Invalid unary operation for type 'T'.
Semantic errors were caught during analysis.
